### PR TITLE
feat(pxp): hookup zephyr interrupts

### PR DESCRIPTION
**Pull request by faxe1008**
### Description of the feature or fix

While adding PXP support for zephyr by adding the semaphore, adding the actual interrupts was missing. It was unclear what the best place to put this is. This adds a depedency on the Zephyr symbol `CONFIG_DMA_INIT_PRIORITY` and the IRQ API, both of which are stable. Originally proposed via: [https://github.com/zephyrproject-rtos/lvgl/pull/52](https://github.com/zephyrproject-rtos/lvgl/pull/52)

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/draw/nxp/pxp/lv_gpu_nxp_pxp_osa.c [hunk 1]
```diff
@@ -141,20 +142,25 @@ static lv_res_t _lv_gpu_nxp_pxp_interrupt_init(void)
...
 
-    NVIC_EnableIRQ(LV_GPU_NXP_PXP_IRQ_ID);
+    s_pxpIdle = true;
 
     return LV_RES_OK;
 }
 
 static void _lv_gpu_nxp_pxp_interrupt_deinit(void)
 {
-    NVIC_DisableIRQ(LV_GPU_NXP_PXP_IRQ_ID);
```
Comment (XuNeo) [inline src/draw/nxp/pxp/lv_gpu_nxp_pxp_osa.c:154]: Do we have other OS in use? Maybe add it to `#else ` is better for backward compatibility.

Comment (faxe1008) [reply src/draw/nxp/pxp/lv_gpu_nxp_pxp_osa.c:154]: Good point, did not consider that this changes behaviour for other OSes

**Comment by kisvegabor** [comment]
cc @anaGrad @nicusorcitu

File: src/draw/nxp/pxp/lv_gpu_nxp_pxp_osa.c (hunk 1 above)
Comment (faxe1008) [reply src/draw/nxp/pxp/lv_gpu_nxp_pxp_osa.c:154]: Applied your suggestion.

**Comment by faxe1008** [comment]
@kisvegabor  Just noticed while setting up a zephyr testcase for this: In order to compile we need to provide the displays dimensions as compile time constants `LCD_WIDTH` and `LCD_HEIGHT`. I assume that this what the `lvgl_support.h` file is for. Would it be feasable to make those Kconfig and lv_conf.h values instead? Something like `LV_NXP_PXP_DISPLAY_WIDTH`?

**Comment by kisvegabor** [comment]
`LCD_WIDTH` and `LCD_HEIGHT` should be the same as `lv_display_get_horizontal/vertical_resolution(disp)`. You can get `disp` by using `_lv_refr_get_disp_refreshing()`. So I can imagine something like:
```c
#ifndef LCD_WIDTH
#define LCD_WIDTH lv_display_get_horizontal_resolution(_lv_refr_get_disp_refreshing())
#endif
``` 

Can you check if it's really working?

**Comment by faxe1008** [comment]
I think using the methods is not really an option since I guess those need to be compile time defined:
[https://github.com/lvgl/lvgl/blob/release/v8.4/src/draw/nxp/pxp/lv_draw_pxp_blend.c#L51C13-L51C30](https://github.com/lvgl/lvgl/blob/release/v8.4/src/draw/nxp/pxp/lv_draw_pxp_blend.c#L51C13-L51C30)

**Comment by nicusorcitu** [comment]
We should remove the PXP temp buffer. It has been removed in lvgl v9, as multiple PXP steps have been disabled. In V9 the PXP will handle only the operations that can be done in a single shot.

I will speak to @cristian-stoica to remove this temp buffer on lvgl v8.4 too, it was a bad decision to add it.
Once we remove it, we no longer need those defines.

**Comment by nicusorcitu** [comment]
Not stale.

**Comment by nicusorcitu** [comment]
@cristian-stoica please help concluding on lvgl v8 (this PR) and lvgl v9: https://github.com/lvgl/lvgl/pull/6298

**Comment by kisvegabor** [comment]
Could you summarize what is the status of the Zephyr + LVGL v9 update?
I get more and more questions about it. It's clear that Zephyr and LVGL is important for our users :rocket:

**Comment by faxe1008** [comment]
I have a PR handy for v9.1, however the monochrome support is still missing. Merging it and basically disabling all downstream projects is not an option.

**Comment by kisvegabor** [comment]
Understood, let's make the indexed 1 bit PR work :slightly_smiling_face:

**Comment by kisvegabor** [comment]
Yesterday I found the issue and added an [example for the fix](https://github.com/lvgl/lvgl/pull/6345#issuecomment-2180214491).
